src/pages/labelPage/LabelPage.jsx

```diff
@@ -1,9 +1,202 @@
 import { Label } from "components/index";
+import { useDocument } from "customHooks/useDocument";
+import "./labelPage.css";
+import { useEffect, useState } from "react";
+import { useNotes } from "context";
 
 function LabelPage() {
+  useDocument("Label");
+
+  const getTime = (time) => {
+    const d = new Date(time);
+    return d.getTime();
+  };
+
+  const [filters, setFilters] = useState({
+    all: true,
+    Personal: false,
+    Work: false,
+    Home: false,
+    lowToHigh: false,
+    highToLow: false,
+    latest: false,
+    oldest: false,
+  });
+
+  const order = { Low: 1, Medium: 2, High: 3 };
+  const labels = ["Home", "Work", "Personal"];
+
+  const {
+    noteState: { notes },
+  } = useNotes();
+
+  const [filteredData, setFilteredData] = useState(notes);
+
+  useEffect(() => {
+    let newData = [...notes];
+    if (filters.lowToHigh) {
+      newData.sort((a, b) => order[a.priority] - order[b.priority]);
+    }
+    if (filters.highToLow) {
+      newData.sort((a, b) => order[b.priority] - order[a.priority]);
+    }
+    if (filters.latest) {
+      newData.sort((a, b) => getTime(b.createdDate) - getTime(a.createdDate));
+    }
+    if (filters.oldest) {
+      newData.sort((a, b) => getTime(a.createdDate) - getTime(b.createdDate));
+    }
+    if (filters.all) {
+      setFilteredData(newData);
+    } else {
+      const selectedLabels = labels.filter((item) => filters[item]);
+      newData = newData.filter((item) => selectedLabels.includes(item.label));
+      setFilteredData(newData);
+    }
+  }, [filters]);
```

Comment:
appreciation - good that you're doing the filteration only if filter state changes